searx/plugins/__init__.py

```diff
@@ -77,6 +87,81 @@ def call(self, ordered_plugin_list, plugin_type, request, *args, **kwargs):
         return ret
 
 
+def load_external_plugins(plugin_names):
+    plugins = []
+    for name in plugin_names:
+        logger.debug('loading plugin: {0}'.format(name))
+        try:
+            pkg = import_module(name)
+        except Exception as e:
```

Comment:
What kind of exception do you are expecting here?  I mean, if there is a syntax error (or any other error) in the imported module .. should we really catch it?  I guess you had some experience where exceptions has been thrown you wanted to catch .. what kind of exception did you have seen?

Reply:
> What kind of exception do you are expecting here? 

Import error

Reply:
I vote for catching explicit exception we can handle or not to catch at all / by example, when we catch:

     CRITICAL:searx.plugins:failed to load plugin module plugin1: name 'y' is not defined

and if we do not catch:

````
Traceback (most recent call last):
  File "./searx/webapp.py", line 74, in <module>
    from searx.search import SearchWithPlugins, get_search_query_from_webapp
  File "./searx/searx/search.py", line 37, in <module>
    from searx.plugins import plugins
  File "./searx/searx/plugins/__init__.py", line 166, in <module>
    plugins.register(*settings['plugins'], external=True)
  File "./searx/searx/plugins/__init__.py", line 67, in register
    plugins = load_external_plugins(plugins)
  File "./searx/searx/plugins/__init__.py", line 94, in load_external_plugins
    pkg = import_module(name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "./searx/local/py3/lib/python3.8/site-packages/plugin1/__init__.py", line 9, in <module>
    x=y
NameError: name 'y' is not defined
````

There could be many reasons why a import fails, bring it to the top so developers have a meaningful error message.

Here is another example which might be worse to catch ModuleNotFoundError ...

````
Traceback (most recent call last):
  File "./searx/webapp.py", line 74, in <module>
    from searx.search import SearchWithPlugins, get_search_query_from_webapp
...
    pkg = import_module(name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named 'plugin2'
````